components/board/collector.go

```diff
@@ -0,0 +1,97 @@
+package board
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"go.viam.com/rdk/data"
+)
+
+type method int64
+
+const (
+	analogs method = iota
+	gpios
+)
+
+func (m method) String() string {
+	if m == analogs {
+		return "Analogs"
+	}
+	if m == gpios {
+		return "Gpios"
+	}
+	return "Unknown"
+}
+
+type analogRecords struct {
```

Comment:
I believe the structs returned by the capture functions need to be public for the `protoutils.StructToStructPb` method to be able to properly convert them to a proto struct. Can't seem to find the repo on github and link, but from the doc string:

```
// StructToStructPb converts an arbitrary Go struct to a *structpb.Struct. Only exported fields are included in the
// returned proto.
```

Reply:
make sense I'll change that before merging